library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
>This is not the case.

If a few records with the same key located both at the end of first index block and at the beginning of subsequent index block, then additional pread required. 

> changes at 413+ lines were added to address this issue, weren't they?

Yes. This lines simply adopts logic that was here in mmap version to pread imlementation.